torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1359,14 +1359,24 @@ def sample_inputs_bmm(self, device, dtype, requires_grad, **kwargs):
     )
 
 def sample_inputs_dot_vdot(self, device, dtype, requires_grad, **kwargs):
-    return (
-        SampleInput(
+    sample_inputs=[]
+    sample_inputs.append(SampleInput(
+        make_tensor((S, ), device, dtype, low=None, high=None, requires_grad=requires_grad),
+        args=(
+            make_tensor((S, ), device, dtype, low=None, high=None, requires_grad=requires_grad),
+        )
+    ))
+    if dtype.is_complex:
+        # dot/vdot for (conj(input), conj(arg_tensor)) and (conj(input), arg_tensor)
+        # is tested in test_conj_view (which tests operations with only conjugated input tensor
+        # -- not conjugated arg tensors)
```

Comment:
Is the forward behaviour tested here?

Reply:
`dot` doesn't have `ref` function, so its correctness is not tested, it seems.

Reply:
yeah all the three cases  `(self.conj(), other.conj())`, `(self.conj(), other.conj())`, `(self, other.conj())` for both forward and backward  will be tested in `test_conj_view`

Reply:
I think going forward it would be good to modify `test_conj_view` so it conjugates the second argument also (at least for binary functions), instead of pushing this complexity to sample inputs. Then, as more operations can take advantage of the fast path, that functionality would be tested  automatically without the need to update sample inputs.

Reply:
fair point!